src/details/ArborX_DetailsDistributor.hpp

```diff
@@ -181,17 +181,38 @@ class Distributor
     requests.reserve(outdegrees + indegrees);
     for (int i = 0; i < indegrees; ++i)
     {
-      requests.emplace_back();
-      MPI_Irecv(src_buffer.data() + _src_offsets[i] * num_packets,
-                _src_counts[i] * num_packets * sizeof(ValueType), MPI_BYTE,
-                _sources[i], MPI_ANY_TAG, _comm, &requests.back());
+      if (_sources[i] != comm_rank)
+      {
+        auto const message_size =
+            _src_counts[i] * num_packets * sizeof(ValueType);
+        auto const receive_buffer_ptr =
+            src_buffer.data() + _src_offsets[i] * num_packets;
+        requests.emplace_back();
+        MPI_Irecv(receive_buffer_ptr, message_size, MPI_BYTE, _sources[i], 123,
```

Comment:
Is this just a good habit to fix the tag here?

Reply:
I am always wary of fixing an MPI tag because you never know what a user will pick and if things will hang if they happen to be the same - typically I try to expose this to the user in an interface for code that does MPI communication with a default value provided

Reply:
> typically I try to expose this to the user in an interface for code that does MPI communication with a default value provided

How do you do that? Do all your communications use the same tag?

Reply:
In this case I would expose in either the distributed tree constructor or send across network - basically any function that does MPI. If you have a single function that does multiple MPI communications which need different tags then you could ask for two or more tags or a range of tags that are acceptable for the library to use. See this for example: https://github.com/ECP-copa/Cabana/blob/89240fe1a605589d7c0d925b41b3b57bc5586459/core/src/Cabana_Distributor.hpp#L100

Reply:
Isn't a much better and cleaner solution to do `MPI_Comm_dup` as that creates a new namespace so that any tag you use won't intersect with user communication? I think in fact this should be done in ArborX.

Reply:
As long as the object is long-lived that sounds reasonable. Either way we shouldn't use a fixed tag with a user-provided communicator. Either the approach I suggested or the approach @aprokop suggested will resolve this.

Reply:
Comm duplication was merged in #135.